### PR TITLE
Simplify code in `Angle`

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

There are no major changes, but there are now fewer branches, fewer local variables, fewer lines of code and lower [McCabe complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity).

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/coordinates/angles/core.py [hunk 1]
```diff
@@ -377,23 +373,16 @@ def to_string(
...
-                    s = f"${s}$"
-                return s
-            s = f"{val}"
-            return s
-
-        values = self.to_value(unit)
-        format_ufunc = np.vectorize(do_format, otypes=["U"])
-        result = format_ufunc(values)
+            if np.isnan(val):
+                return "nan"
```
Comment (mhvk) [inline astropy/coordinates/angles/core.py:377]: This is not quite right if the `Angle` is complex - though really the rest of the code breaks for that case too... so, for this refactoring PR it doesn't matter. 

Aside: It is also slightly bizarre that no `$` are used for latex, but that's even more out of scope here. Really, we need to use `Quantity.to_string()`...

File: astropy/coordinates/angles/core.py [hunk 2]
```diff
@@ -514,21 +503,17 @@ def is_within_bounds(self, lower=None, upper=None):
...
             `True` if all angles satisfy ``lower <= angle < upper``
         """
-        ok = True
-        if lower is not None:
-            ok &= np.all(Angle(lower) <= self)
-        if ok and upper is not None:
-            ok &= np.all(self < Angle(upper))
-        return bool(ok)
+        # Avoid `any()` so that a NaN value would return `False`
+        if lower is not None and not (Angle(lower) <= self).all():
```
Comment (mhvk) [inline astropy/coordinates/angles/core.py:507]: Maybe combine all?
```
return (lower is None or np.all(self >= Angle(lower))) and (upper is None or np.all(self < Angle(upper)))
```

**Review by mhvk — COMMENTED** [review]
Thanks, a few small comments.

File: astropy/coordinates/angles/core.py (hunk 2 above)
Comment (eerovaher) [reply astropy/coordinates/angles/core.py]: I wrote the comment to warn against replacing `not (Angle(lower) <= self).all()` with `(Angle(lower) > self).any()`, but with your suggestion the comment is not needed anymore.

**Review by mhvk — APPROVED** [review]
Thanks, looks all good now!

**Pull request merged**